Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/ActionsTreeView.cs

```diff
@@ -233,9 +233,24 @@ internal void AddComposite(string actionName, string compositeType)
         private void DeleteItem(ActionOrBindingData data)
         {
             if (data.isAction)
+            {
+                string actionToSelect = GetPreviousActionNameFromViewTree(data);
                 Dispatch(Commands.DeleteAction(data.actionMapIndex, data.name));
+                Dispatch(Commands.SelectAction(actionToSelect));
+            }
             else
+            {
+                int bindingIndexToSelect = GetPreviousBindingIndexFromViewTree(data, out string parentActionName);
                 Dispatch(Commands.DeleteBinding(data.actionMapIndex, data.bindingIndex));
+
+                if (bindingIndexToSelect >= 0)
+                    Dispatch(Commands.SelectBinding(bindingIndexToSelect));
+                else
+                    Dispatch(Commands.SelectAction(parentActionName));
+            }
+
+            // Deleting an item sometimes causes the UI Panel to lose focus; make sure we keep it
+            m_ActionsTreeView.Focus();
```

Comment:
I wanted to avoid combining Delete/Select since I think they should be individual actions, but that was before discovering the UI refreshed between them. Maybe instead there's a way to "suspend" UI updates when executing multiple commands?

I do agree it's a minor issue and something can revisit later.

Reply:
I think you can call the select command inside the delete command, what would mean we still have a delete and select command.

Reply:
I played around with this a bit more to try and "combine" the Select/Delete operations and avoid needing to call `Focus()` on the TreeView. My idea was to create a "CommandList" implementation, that executed multiple commands (in sequence) and only fire the `StateChanged` event once at the end. [This commit](f7e317d1f10cccbff13dbe9d9616e909127e1521) holds my quick, dirty prototype.

Unfortunately, it didn't work; deleting multiple Action/Bindings in a row will cause the TreeView to (somehow) lose focus. I also found this issue exists in `develop` without my changes, and so It appears to be a deeper issue (possibly a bug in UITK).

So I'm inclined to keep the `Focus()` call in to work-around the problem.

Reply:
Interesting! Thank you for checking, I guess we are good to keep the Focus() work-around for now.

Reply:
I opened[ ticket ISX-1631](https://jira.unity3d.com/browse/ISX-1633) to follow up with this later.
It could be related to some other "focusing" issues.